tableauserverclient/models/user_item.py

```diff
@@ -260,4 +256,157 @@ def _parse_element(user_xml, ns):
         )
 
     def __repr__(self) -> str:
-        return "<User {} name={} role={}>".format(self.id, self.name, self.site_role)
+        str_site_role = self.site_role or "None"
+        return "<User {} name={} role={}>".format(self.id, self.name, str_site_role)
+
+    # valid values for each field
+    CHOICES: List[List[str]] = [
+        [],
+        [],
+        [],
+        ["creator", "explorer", "viewer", "unlicensed"],  # license
+        ["system", "site", "none", "no"],  # admin
+        ["yes", "true", "1", "no", "false", "0"],  # publisher
+        [],
+        [Auth.SAML, Auth.OpenID, Auth.ServerDefault],  # auth
+    ]
+
+    class CSVImportFileItem(object):
+
+        # CSV import file format
+        # username, password, display_name, license, admin_level, publishing, email, auth type
+        class Column(IntEnum):
+            USERNAME = 0
+            PASS = 1
+            DISPLAY_NAME = 2
+            LICENSE = 3  # aka site role
+            ADMIN = 4
+            PUBLISHER = 5
+            EMAIL = 6
+            AUTH = 7
+
+            MAX = 7
+
+        @staticmethod
+        def _create_from_csv_line(line: str):
+            if line is None or line is False or line == "\n" or line == "":
+                return None
+            line = line.strip().lower()
+            values: List[str] = list(map(str.strip, line.split(",")))
+            user = UserItem(values[UserItem.CSVImportFileItem.Column.USERNAME])
+            if len(values) > 1:
+                if len(values) > UserItem.CSVImportFileItem.Column.MAX:
+                    raise ValueError("Too many attributes for user import")
+                while len(values) <= UserItem.CSVImportFileItem.Column.MAX:
+                    values.append("")
+                site_role = UserItem.CSVImportFileItem.evaluate_site_role(
+                    values[UserItem.CSVImportFileItem.Column.LICENSE],
+                    values[UserItem.CSVImportFileItem.Column.ADMIN],
+                    values[UserItem.CSVImportFileItem.Column.PUBLISHER],
+                )
+
+                user._set_values(
+                    None,
+                    values[UserItem.CSVImportFileItem.Column.USERNAME],
+                    site_role,
+                    None,
+                    None,
+                    values[UserItem.CSVImportFileItem.Column.DISPLAY_NAME],
+                    values[UserItem.CSVImportFileItem.Column.EMAIL],
+                    values[UserItem.CSVImportFileItem.Column.AUTH],
+                    None,
+                )
+            return user
+
+        @staticmethod
+        def validate_file_for_import(csv_file: io.TextIOWrapper, logger) -> List[int]:
+            num_errors = 0
```

Comment:
I know the code and function names are descriptive but it would be nice to have comments still about what the functions are doing.

Reply:
Added comments